app/components/markdown/markdown_image/markdown_image.js

```diff
@@ -85,10 +86,11 @@ export default class MarkdownImage extends React.Component {
         return null;
     }
 
-    componentDidUpdate(prevProps) {
+    async componentDidUpdate(prevProps) {
         if (this.props.source !== prevProps.source) {
             // getSource also depends on serverURL, but that shouldn't change while this is mounted
-            ImageCacheManager.cache(null, this.getSource(), this.setImageUrl);
+            const source = await this.getSource();
+            ImageCacheManager.cache(null, source, this.setImageUrl);
```

Comment:
perhaps best to move this to `getSource` and remove the async/await in componentDidUpdate? also maybe we should make sure we do not need to fetch the image cache every time the component updates? I'm worried this actually is causing multiple updates